modules/extreme_year.py

```diff
@@ -0,0 +1,117 @@
+"""
+this module will return the highest and lowest temperatures of a
+given year along with the most humid day.
+"""
+import glob
+
+
+def read_and_calculate_extremes(pattern, path):
+    """
+    this function reads all the files at the given path which contain the given pattern
+    from each file that matches the pattern finds the following:
+    1. max highest temperature
+    2. max lowest temperature
+    3. max humidity
+    :param pattern: a string which contains a year e.g '2006', '2007'
+    :param path: a string which contains path to a directory containing weather files
+    :return: a 3x2 list maximums where
+    maximums[0] = [max_highest_temperature, date]
+    maximums[1] = [max_lowest_temperature, date]
+    maximums[2] = [max_humidity, date]
+    date here is a string like: '2002-4-1', '2004-3-14'
+    """
+    files = glob.glob(path + f"*{pattern}*")
+    if not files:
+        print(
+            f"We don't have information regarding the "
+            f"year {pattern} in the given path"
+        )
+        return -1
+    # maximums[0] contains highest temperature and day
+    # maximums[1] contains lowest temperature and day
+    # maximums[2] contains most humidity and day
+    maximums = [[0, ""], [0, ""], [0, ""]]
+    initialized = False
+    # indexes to compare values with.
+    # highest temperature is stored at index 1 of weather reading
+    # lowest temperature is stored at index 3 of weather reading
+    # max humidity is stored at index 7 of weather reading
+    # see the weather files for further explanation
+    indexes = [1, 3, 7]
+    for i in files:
+        with open(i, "r") as file:
+            # skip first line as it contains field names
+            file.readline()
+            for line in file:
+                # since readline() returns 'linedata\n' with split by '\n' to get ['linedata','\n]
+                # then we get 'linedata' by grabbing the [0]
+                # lastly,  we split by ',' to get the entries parsed
+                parsed_line = line.split("\n")[0].split(",")
+                if not initialized:
+                    for j, val in enumerate(indexes):
+                        maximums[j] = [int(parsed_line[val]), parsed_line[0]]
+                    initialized = True
+                    continue
+                for j, val in enumerate(indexes):
+                    # index of weather record to compare value of current maximum with
+                    index_to_compare = val
+                    # value of temperatures or humidity or none
+                    value = parsed_line[index_to_compare]
```

Comment:
When you know the index and you have a line read already, can you try creating generic methods e.g. `get_temperature(temperature_col_index)`, `get_humidity(humitity_col_index)` etc.

Reply:
i did not understand this comment. should i pass my line read and index here and get the value?

Reply:
My bad, I think we already know the index where we can find specific value e.g. temperature and humidity, In which case we might not need to pass the index as that is going to remain constant for all the weather files. We'd only need to pass the line.

Reply:
done